sogs/routes/users.py

```diff
@@ -76,16 +76,89 @@ def extract_rooms_or_global(req, admin=True):
     return (None, True)
 
 
-def _serialize_message(msg):
+def _serialize_message(msg, *, key):
     return {
         "id": msg.id,
         "posted_at": msg.posted_at,
         "expires_at": msg.expires_at,
         "message": utils.encode_base64(msg.data),
-        "sender": msg.sender.session_id,
+        key: getattr(msg, key).session_id,
     }
 
 
+def _box(func, **kwargs):
+    """ handle inbox/outbox endpoints common logic """
+    if g.user.banned or not g.user.is_blinded:
+        abort(http.FORBIDDEN)
+    keys = {'to': 'sender', 'sent': 'recipient'}
+    key = keys.get(func)
+    limit = utils.get_int_param('limit', 100, min=1, max=256, truncate=True)
+    get = getattr(Message, func)
+    msgs = [_serialize_message(msg, key=key) for msg in get(user=g.user, limit=limit, **kwargs)]
+    if msgs or not kwargs:
+        return jsonify(msgs)
+    return Response('', status=http.NOT_MODIFIED)
```

Comment:
This indirection with `func` and key as strings that goes through a dict lookup makes this a bit harder to follow than necessary.  If this were just a bool it would be clearer:

```Python
def _box(in:bool=True, **kwargs):
    # ...
    get = Message.to if in else Message.sent
    msgs = [_serialize_message(msg) for msg in get(user=g.user, limit=limit, **kwargs)]
```